### PR TITLE
Update Debugging_General.mdx

### PR DESCRIPTION
**Pull request by aksh1**

### CODE REVIEW

File: content/1_General/Debugging_General.mdx [hunk 1]
```diff
@@ -22,13 +22,25 @@ This module is based on both of these resources. I've included the content that
...
+   can be simplified to...
+   ```
+   private static int nextInt() {
+   	Integer.parseInt(stringTokenizer.nextToken());
+   }
+   
+   public static void main(String[] args) throws IOException {
+   	int n = nextInt();
+	int k = nextInt();
+   }
```
Comment (HarryWangATX) [inline content/1_General/Debugging_General.mdx:42]: idt IO is a good example for this maybe like doing some operations??

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: content/1_General/Debugging_General.mdx (hunk 1 above)
Comment (aksh1) [reply content/1_General/Debugging_General.mdx:42]: Ok so like maybe swapping items in an array?

File: content/1_General/Debugging_General.mdx [hunk 2]
```diff
@@ -23,22 +23,33 @@ This module is based on both of these resources. I've included the content that
...
+	arr[j] = first;
    }
    
-   public static void main(String[] args) throws IOException {
-   	int n = nextInt();
-	int k = nextInt();
+   public static void main(String[] args) {
+   	swap(0, 2, arr);
+	swap(1, 3, arr);
    }
```
Comment (HarryWangATX) [inline content/1_General/Debugging_General.mdx:54]: close with ```

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: content/1_General/Debugging_General.mdx [hunk 3]
```diff
@@ -22,13 +22,36 @@ This module is based on both of these resources. I've included the content that
...
 
-<IncompleteSection>
-
-todo - find particularly bad example of repetition
-
-</IncompleteSection>
-
  - Don't repeat yourself!
+   - Consider this example of swapping items in an array of integers. 
+   ```
```
Comment (frodakcin) [inline content/1_General/Debugging_General.mdx:27]:
```suggestion
   ```java
```

File: content/1_General/Debugging_General.mdx [hunk 4]
```diff
@@ -22,13 +22,36 @@ This module is based on both of these resources. I've included the content that
...
+	   
+	   //swap second item with fourth
+	   int second = arr[1];
+	   int secondLast = arr[3];
+	   arr[1] = secondLast;
+	   arr[3] 
+   }
+   ```
+   Hopefully you notice the unnecessary repetition in this code. Luckily, we can turn this into a function that takes in the two indicies of the array that we want to swap, like so.
+   ```
```
Comment (frodakcin) [inline content/1_General/Debugging_General.mdx:43]:
```suggestion
   ```java
```

File: content/1_General/Debugging_General.mdx [hunk 5]
```diff
@@ -22,13 +22,36 @@ This module is based on both of these resources. I've included the content that
...
+	   int last = arr[2];
+	   arr[0] = last;
+	   arr[2] = first;
+	   
+	   //swap second item with fourth
+	   int second = arr[1];
+	   int secondLast = arr[3];
+	   arr[1] = secondLast;
+	   arr[3] 
+   }
```
Comment (frodakcin) [inline content/1_General/Debugging_General.mdx:40]: use all tabs

File: content/1_General/Debugging_General.mdx (hunk 3 above)
Comment (frodakcin) [reply content/1_General/Debugging_General.mdx:27]: Also, since this is a java example, wrap in LanguageSection & JavaSection?

**Review by frodakcin — CHANGES_REQUESTED** [review]

**Review by frodakcin — APPROVED** [review]
I think it's ok.

Some possible problems:
 - Java code uses tabs, but the bullet points use spaces
 - CPP/Python can't use this example (they both have builtin swap functions)

**Comment by thecodingwizard** [comment]
@bqi343 did you have a particular example in mind when you wrote 

> todo - find particularly bad example of repetition

File: content/1_General/Debugging_General.mdx [hunk 6]
```diff
@@ -22,13 +22,49 @@ This module is based on both of these resources. I've included the content that
...
+	int first = arr[0];
+	int last = arr[2];
+	arr[0] = last;
+	arr[2] = first;
+
+	//swap second item with fourth
+	int second = arr[1];
+	int secondLast = arr[3];
+	arr[1] = secondLast;
+	arr[3];
```
Comment (bqi343) [inline content/1_General/Debugging_General.mdx:45]: ^ what's this line supposed to do ...

**Review by bqi343 — CHANGES_REQUESTED** [review]
tbh I don't think this is a particularly bad example of repetition unless you're doing a lot more swaps. I think

```java
int tmp;
tmp = arr[2]; arr[2] = arr[0]; arr[0] = tmp;
tmp = arr[3]; arr[3] = arr[1]; arr[1] = tmp;
```

is fine?

**Comment by bqi343** [comment]
I think my tttt solution is a big improvement over Jesse's, I guess that could be an example :P 

https://github.com/cpinitiative/usaco-guide/pull/454/commits/e3d1c0cbdfa6cbcd07b56987ae591eab2a2608af
https://usaco.guide/solutions/usaco-831?lang=cpp

File: content/1_General/Debugging_General.mdx (hunk 6 above)
Comment (frodakcin) [reply content/1_General/Debugging_General.mdx:45]:
```suggestion
	arr[3] = second;
```

**Review by darren-yao — APPROVED** [review]

**Pull request merged**